scripts/zowe-install-extension.sh

```diff
@@ -0,0 +1,148 @@
+
+    if [ "$EXT_COMP_VAL" = " " ]; then
+        sed -e "s/EXTERNAL_COMPONENTS=/EXTERNAL_COMPONENTS=$EXTENSION_BIN_PATH/" $INSTANCE_DIR/instance.env > $INSTANCE_DIR/instance.env.tmp
+        mv $INSTANCE_DIR/instance.env.tmp $INSTANCE_DIR/instance.env
+    else
+        # Ensures that the bin directory of the extension is included into the instance.env once (Avoids duplication if same extension is installed twice)
+        if [[ $(grep 'EXTERNAL_COMPONENT' $INSTANCE_DIR/instance.env | grep $TARGET_DIR/$EXTENSION_NAME/bin) = "" ]]; then
+            sed -e "s/EXTERNAL_COMPONENTS=/EXTERNAL_COMPONENTS=$EXTENSION_BIN_PATH,/" $INSTANCE_DIR/instance.env > $INSTANCE_DIR/instance.env.tmp
+            mv $INSTANCE_DIR/instance.env.tmp $INSTANCE_DIR/instance.env
+        fi
+    fi
+
+    echo "Zowe component has been installed."
+}
+
+install_desktop_plugin(){
+    cd $TARGET_DIR
+
+    if [ ! -d "$EXTENSION_NAME" ]; then
+        mkdir $EXTENSION_NAME
+    else
+        # Clean up the directory and make a new one
+        rm -rf $EXTENSION_NAME
+        mkdir $EXTENSION_NAME
+    fi
+
+    cd $EXTENSION_NAME
+
+    pax -ppx -rf "$EXTENSION_PATH"
+
+    # Validates the pax file is correctly configured when unpaxed
+    if [ ! -e "pluginDefinition.json" ]; then
+        error_handler "$EXTENSION_PATH: Invalid package"
+    fi
+
+    # Uses install-app.sh in zowe-instance-dir to automatically set up the extension onto zowe
+    $INSTANCE_DIR/bin/install-app.sh $TARGET_DIR/$EXTENSION_NAME
+
+    echo "Zowe desktop plugin has been installed."
+}
+
+while [ $# -gt 0 ]; do #Checks for parameters
+    arg="$1"
+    case $arg in
+        -e|--extension) #Represents the path pointed to the extension's pax file
+        shift
+        if [[ "$1" = *.pax ]]; then
+            EXTENSION_PATH=$1
+        else
+            error_handler "-e|--extension: Given path is not a pax file or does not exist"
+        fi
+        shift
+        ;;
+        -i|--instance_dir) #Represents the path to zowe's instance directory 
+        shift
+        if [ -d "$1" ] && [ -e "$1/instance.env" ]; then
+            INSTANCE_DIR=$1
+        else
+            error_handler "-i|--instance_dir: Given path is not a zowe instance directory or does not exist"
+        fi
+        shift
+        ;;
+        -d|--target_dir)
+        shift
+        TARGET_DIR=$1
+        shift
+        ;;
+        -n|--name) #Represents the name of the plugin/extension
+        shift
+        EXTENSION_NAME="$1"
+        shift
+        ;;
+        -t|--type) #Represents the type of extension whether its a desktop plugin or web component
+        shift
+        if [[ $1 == "component" ]] || [[ $1 == "desktop-plugin" ]]; then
+            INSTALL_TYPE=$1
+        else
+            error_handler "-t|--type: Invalid value, possible values are 'component' and 'desktop-plugin'"
+        fi
+        shift
+        ;;
+        *)
```

Comment:
In addition to the original design, would like to add one more set of `-l|--logs-dir` to write installation logs to it. The value is optional, default will be `<instance-dir>/logs`. The log file should look like `install-extension-2020-09-24-13-00-26.log`.

In the log file, it should shows all the command line parameters, and all the steps including failures. We can follow other log as example.

Try to use `bin/utils/setup-log-dir.sh` and follow other scripts how the logs are written.